protostar-rust/src/lib.rs

```diff
@@ -0,0 +1,59 @@
+use pyo3::prelude::*;
+use pyo3::wrap_pyfunction;
+use pyo3::exceptions::RuntimeError;
+
+use anyhow::Context;
+use std::collections::HashMap;
+
+use cairo_lang_protostar::test_collector::collect_tests;
+use cairo_lang_runner::{SierraCasmRunner, RunResultValue};
+
+fn run_result_value_to_string(run_result: RunResultValue) -> String {
+  match run_result {
+    RunResultValue::Success(data) => {
+      return format!("Success {:?}", data);
+    }
+    RunResultValue::Panic(data) => {
+        
+      return format!("Panic {:?}", data);
+    }
+  }
+}
+
+fn internal_run_tests(input_path: &str) -> anyhow::Result<Vec<(String, String)>> {
+  let (sierra_program, test_configs) = collect_tests(&input_path.to_owned(), None, None, None)?;
+
+  let runner = SierraCasmRunner::new(
+    sierra_program,
+    None,
+    HashMap::new(),
+  )
+  .with_context(|| "Failed setting up runner.")?;
+
+  let mut results = vec![];
+  for config in &test_configs {
+    let result = runner
+    .run_function(
+        runner.find_function(&config.name.as_str())?,
+        &[],
+        config.available_gas,
+        Default::default(),
+    )
+    .with_context(|| format!("Failed to run the function `{}`.", config.name.as_str()))?;
+    results.push((config.name.clone(), run_result_value_to_string(result.value)));
```

Comment:
Can we print results here instead of returning them?

Reply:
Hmm, yes, right, once we start supporting parallel tests, we should print the results in Rust anyway so we can start doing it right away. Thanks for pointing this out.

Reply:
[add printing](https://github.com/software-mansion/protostar/pull/1881/commits/bfad1b5d5dfa999cd6b78cfceb8d69f9bcdf2907)

Reply:
> Hmm, how about printing and returning them? The point in returning is that we can easily test the outputs in protostar.

I think we shouldn't have such tests on the protostar side.

Reply:
discussed offline
[do not return value from the binding](https://github.com/software-mansion/protostar/pull/1881/commits/99554b3018b98a4d792ac7c82575b14dcacf9467)